components/support/ktx/src/main/java/mozilla/components/support/ktx/android/net/Uri.kt

```diff
@@ -53,3 +57,22 @@ fun Uri.sameSchemeAndHostAs(other: Uri) = scheme == other.scheme && host == othe
  * https://html.spec.whatwg.org/multipage/origin.html#same-origin
  */
 fun Uri.sameOriginAs(other: Uri) = sameSchemeAndHostAs(other) && port == other.port
+
+/**
+ * Indicate if the [this] uri is under the application private directory.
+ */
+fun Uri.isUnderPrivateAppDirectory(context: Context): Boolean {
```

Comment:
Yes, that's correct. I would like to keep it. In case we introduce support in the future, and miss this check.  I comment could clarify why it's there. What do you think?

Reply:
Then I see no problem in adding it. The logic seems okay 👍

Reply:
👍

Reply:
I second @Amejia481's comment about including this. Apps other than Fenix can consume the AC project. GV supports down to API 19 so there are still usecases where `file://` could be passed to AC.

Reply:
That is in fact a valid use case. However, as I said, AC still can't process those Uris. Unless they change how `getFileName` calculates a file name. Because, otherwise, AC will try to load a cursor from a `file://` which will just fail.